presto-main/src/test/java/com/facebook/presto/sql/TestExpressionInterpreter.java

```diff
@@ -686,6 +686,8 @@ public void testCastOptimization()
         assertOptimizedEquals("cast(bound_integer + 1 as VARCHAR)", "'1235'");
         assertOptimizedEquals("cast(bound_long + 1 as VARCHAR)", "'1235'");
         assertOptimizedEquals("cast(unbound_string as VARCHAR)", "cast(unbound_string as VARCHAR)");
+        assertOptimizedMatches("cast(unbound_string as VARCHAR)", "unbound_string");
```

Comment:
I would also add tests with type only coercion and make sure it's not removed for unbound_string, e.g:
```
assertOptimizedMatches("cast(unbound_string as VARCHAR(10))", "cast(unbound_string as VARCHAR(10))");
```